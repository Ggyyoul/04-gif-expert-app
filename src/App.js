import { useState } from 'react';
import { AddCategory } from './components/AddCategory/AddCategory';
import { GifGrid } from './components/GifGrid/GifGrid';

export const App = ( {defaultCategories = []}) => {
  const [categories, setCategories] = useState(defaultCategories);

  return (
    <>
      <h2>Gif Expert App</h2>
      <AddCategory setCategories={setCategories} />
      <hr />
      <ol>
        {categories.map((category) => {
          return <GifGrid key={category} category={category} />;
        })}
      </ol>
    </>
  );
};
