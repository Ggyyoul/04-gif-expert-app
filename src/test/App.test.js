import { shallow } from 'enzyme'
import { App } from "../App"

describe('Test in component <App />', () => {

  test('should make snapshot', () => {

    const wrapper = shallow(<App />)
    expect(wrapper).toMatchSnapshot()
  })

  test('should show a list of categories', () => {

    const categories = ['Twice', 'Seventeen'];
    const wrapper = shallow(<App defaultCategories = {categories} />)

    expect(wrapper).toMatchSnapshot();
    expect(wrapper.find('GifGrid').length).toBe(categories.length)
  })
  
})
