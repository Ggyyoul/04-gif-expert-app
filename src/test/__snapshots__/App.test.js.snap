// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <App /> should make snapshot 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Test in component <App /> should show a list of categories 1`] = `
<Fragment>
  <h2>
    Gif Expert App
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="Twice"
      key="Twice"
    />
    <GifGrid
      category="Seventeen"
      key="Seventeen"
    />
  </ol>
</Fragment>
`;
